hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodiePayloadConfig.java

```diff
@@ -60,6 +54,14 @@ public class HoodiePayloadConfig extends HoodieConfig {
         + "the record payload class to merge records in the log against each other, merge again with the base file and "
         + "produce the final record to be written after compaction.");
 
+  /** @deprecated Use {@link HoodieWriteConfig#PRECOMBINE_FIELD_NAME} and its methods instead */
```

Comment:
would n't the payload config be used on the read/query side as well? does not feel good to have this depend on writeConfig?

Reply:
The pre-combine field is written to table configs in `hoodie.properties` and the read side uses that already.  So is this ok?